src/libs/TransactionPreviewUtils.ts

```diff
@@ -0,0 +1,326 @@
+    }
+    if (comparisonResult.change.tag) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_TAG_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.description) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_DESCRIPTION_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.taxCode) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_TAX_CODE_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.billable) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_BILLABLE_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.reimbursable) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_REIMBURSABLE_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+
+    return ROUTES.TRANSACTION_DUPLICATE_CONFIRMATION_PAGE.getRoute(route.params?.threadReportID, backTo);
+};
+
+type TranslationPathOrText = {
+    translationPath?: TranslationPaths;
+    text?: string;
+};
+
+const dotSeparator: TranslationPathOrText = {text: ` ${CONST.DOT_SEPARATOR} `};
+
+function getTransactionPreviewTextAndTranslationPaths({
+    iouReport,
+    transaction,
+    action,
+    violations,
+    transactions,
+    isBillSplit,
+    shouldShowRBR,
+    violationMessage,
+}: {
+    iouReport: OnyxEntry<OnyxTypes.Report>;
+    transaction: OnyxEntry<OnyxTypes.Transaction>;
+    action: OnyxTypes.ReportAction;
+    violations: OnyxTypes.TransactionViolations;
+    transactions: Partial<TransactionDetails>;
+    isBillSplit: boolean;
+    shouldShowRBR: boolean;
+    violationMessage?: string;
+}) {
+    const isFetchingWaypoints = isFetchingWaypointsFromServer(transaction);
+    const isTransactionOnHold = isOnHold(transaction);
+    const isTransactionMadeWithCard = isCardTransaction(transaction);
+    const isMoneyRequestSettled = isSettled(iouReport?.reportID);
+    const isSettlementOrApprovalPartial = !!iouReport?.pendingFields?.partial;
+    const isPartialHold = isSettlementOrApprovalPartial && isTransactionOnHold;
+    // We don't use isOnHold because it's true for duplicated transaction too and we only want to show hold message if the transaction is truly on hold
+    const shouldShowHoldMessage = !(isMoneyRequestSettled && !isSettlementOrApprovalPartial) && !!transaction?.comment?.hold;
+    const showCashOrCard: TranslationPathOrText = {translationPath: isTransactionMadeWithCard ? 'iou.card' : 'iou.cash'};
+    const isScanning = hasReceipt(transaction) && isReceiptBeingScanned(transaction);
+    const hasFieldErrors = hasMissingSmartscanFields(transaction);
+    const hasViolationsOfTypeNotice = hasNoticeTypeViolation(transaction?.transactionID, violations, true) && isPaidGroupPolicy(iouReport);
+
+    const {amount: requestAmount, currency: requestCurrency} = transactions;
+
+    let RBRmessage: TranslationPathOrText | undefined;
```

Comment:
I think this is correct naming style
```suggestion
    let RBRMessage: TranslationPathOrText | undefined;
```